build.rs

```diff
@@ -136,7 +137,34 @@ fn build_ossl() {
         .expect("Couldn't write bindings!");
 }
 
+#[cfg(feature = "dynamic")]
+fn use_system_ossl() {
+    println!("cargo:rustc-link-lib=crypto");
+    bindgen::Builder::default()
+        .header("ossl.h")
+        .clang_arg("-std=c90") /* workaround [-Wimplicit-int] */
+        .derive_default(true)
+        .formatter(bindgen::Formatter::Prettyplease)
+        .allowlist_item("ossl_.*")
+        .allowlist_item("OSSL_.*")
+        .allowlist_item("openssl_.*")
+        .allowlist_item("OPENSSL_.*")
+        .allowlist_item("CRYPTO_.*")
+        .allowlist_item("c_.*")
+        .allowlist_item("EVP_.*")
+        .allowlist_item("evp_.*")
+        .allowlist_item("BN_.*")
+        .allowlist_item("LN_aes.*")
```

Comment:
Can this list be somehow reused from the non-dynamic linking as it is exactly same and keeping them in sync is quite error-prone?

Reply:
Yeah, let me add an auxiliary function for that